Update gbfs.md (#213)
MobilityData/gbfs: gbfs.md

* [Revision History](#revision-history)
* [Introduction](#introduction)

> * [Version endpoints](#version-endpoints)

Reviewer: ```suggestion
* [Version Endpoints](#version-endpoints)
```

* Provide the status of the system at this moment
* Do not provide information whose primary purpose is historical

The specification supports real-time travel advice in GBFS-consuming applications.

## Version endpoints
The version of the GBFS specification to which a feed conforms is declared in the `version` field in all files. See [Output Format](#output-format).<br />

GBFS Best Practice defines that:<br />

> _GBFS producers_ should provide endpoints that conform to both the current specification long term support (LTS) branch as well as the latest release branch within at least 3 months of a new spec `MAJOR` or `MINOR` version release. It is not necessary to support more than one MINOR release of the same `MAJOR` release group because `MINOR` releases are backwards-compatible. See [specification versioning](https://github.com/NABSA/gbfs/blob/master/README.md#specification-versioning)<br />

Reviewer: ```suggestion
_GBFS producers_ should provide endpoints that conform to both the current specification long term support (LTS) branch as well as the latest release branch within at least 3 months of a new spec `MAJOR` or `MINOR` version release. It is not necessary to support more than one `MINOR` release of the same `MAJOR` release group because `MINOR` releases are backwards-compatible. See [specification versioning](https://github.com/NABSA/gbfs/blob/master/README.md#specification-versioning)<br />
```

* Pagination is not supported.

### File Distribution
* If the publisher intends to distribute as individual HTTP endpoints then:
    * Required files must not 404. They should return a properly formatted JSON file as defined in [Output Format](#output-format).
    * Optional files may 404. A 404 of an optional file should not be considered an error.
* Auto-Discovery:
    * This specification supports auto-discovery.
    * The location of the auto-discovery file will be provided in the HTML area of the shared mobility landing page hosted at the URL specified in the URL field of the system_infomation.json file.
    * This is referenced via a _link_ tag with the following format:
      * `<link rel="gbfs" type="application/json" href="https://www.example.com/data/gbfs.json" />`
    * References:
      * http://microformats.org/wiki/existing-rel-values
      * http://microformats.org/wiki/rel-faq#How_is_rel_used

### Localization

* Each set of data files should be distributed in a single language as defined in system_information.json.
* A system that wants to publish feeds in multiple languages should do so by publishing multiple distributions, such as:

>     * https://www.example.com/data/en/system_information.json

Reviewer: ```suggestion
    * `https://www.example.com/data/en/system_information.json`
```

>     * https://www.example.com/data/fr/system_information.json

Reviewer: ```suggestion
    * `https://www.example.com/data/fr/system_information.json`
```

ttl                 | Yes       | Integer representing the number of seconds before the data in this feed will be updated again (0 if the data should always be refreshed)
version *(beta)*    | Yes       | String - GBFS version number to which the feed confirms, according to the versioning framework.
data                | Yes       | JSON hash containing the data fields for this response
Field Name          | Required  |  Type                |  Defines
--------------------| ----------| -------------------  | ------------------
last_updated        | Yes       |  Timestamp           | Last time the data in the feed was updated.
ttl                 | Yes       | Non-negative integer | Number of seconds before the data in the feed will be updated again (0 if the data should always be refreshed).
version *(beta)*    | Yes       | String               | GBFS version number to which the feed confirms, according to the versioning framework.
data                | Yes       | Object               | Response data in the form of name:value pairs.

Example:
```json
```
{
  "last_updated": 1434054678,
  "ttl": 3600,

>   "version": "1.1",

Reviewer: Undo.

gbfs_versions.json *(beta)* | Optional                | Lists all feed endpoints published according to versions of the GBFS documentation.
system_information.json     | Yes                     | Details including system operator, system location, year implemented, URL, contact info, time zone.
station_information.json    | Conditionally required  | List of all stations, their capacities and locations. Required of systems utilizing docks.
station_status.json         | Conditionally required  | Number of available bikes and docks at each station and station availability. Required of systems utilizing docks.
free_bike_status.json       | Conditionally required  | Bikes that are available for rent. Required of systems that offer bikes for rent outside of stations.
system_hours.json           | Optional                | Hours of operation for the system.
system_calendar.json        | Optional                | Dates of operation for the system.
system_regions.json         | Optional                | Regions the system is broken up into.

> system_pricing_plans.json   | Optional                | The system pricing scheme.

Reviewer: ```suggestion
system_pricing_plans.json   | Optional                | System pricing scheme.
```
Might as well make this consistent with others - had missed it earlier.

system_alerts.json          | Optional                | Current system alerts.

## File Requirements
* All files should be valid JSON.
* All data should be UTF-8 encoded.
* Line breaks should be represented by unix newline characters only (\n).
* Pagination is not supported.

### File Distribution
* If the publisher intends to distribute as individual HTTP endpoints then:
    * Required files must not 404. They should return a properly formatted JSON file as defined in [Output Format](#output-format).
    * Optional files may 404. A 404 of an optional file should not be considered an error.
* Auto-Discovery:
    * This specification supports auto-discovery.

>     * The location of the auto-discovery file will be provided in the HTML area of the bikeshare landing page hosted at the URL specified in the URL field of the system_infomation.json file.

Reviewer: ```suggestion
    * The location of the auto-discovery file will be provided in the HTML area of the shared mobility landing page hosted at the URL specified in the URL field of the system_infomation.json file.
```
Update this to be more inclusive.

>     * This is referenced via a link tag with the following format:

Reviewer: ```suggestion
    * This is referenced via a _link_ tag with the following format:
```

      * `<link rel="gbfs" type="application/json" href="https://www.example.com/data/gbfs.json" />`

>     * Reference:

Reviewer: ```suggestion
    * References:
```

      * http://microformats.org/wiki/existing-rel-values
      * http://microformats.org/wiki/rel-faq#How_is_rel_used

### Localization

* Each set of data files should be distributed in a single language as defined in system_information.json.
* A system that wants to publish feeds in multiple languages should do so by publishing multiple distributions, such as:
    * https://www.example.com/data/en/system_information.json
    * https://www.example.com/data/fr/system_information.json

## Field Types

* Array - A JSON element consisting of an ordered sequence of zero or more values.
* Boolean - One of two possible values, 1= true and 0= false. Boolean values must be JSON booleans, not strings (i.e. true or false, not "true" or "false")

> * Date - Service day in the YYYY-MM-DD format. Example: 2019-09-13 for September 13th, 2019.

Reviewer: ```suggestion
* Date - Service day in the YYYY-MM-DD format. _Example: 2019-09-13 for September 13th, 2019._
```

> * Email - An email address. Example: example@example.com

Reviewer: ```suggestion
* Email - An email address. _Example: example@example.com_
```

* Enum (Enumerable values) - An option from a set of predefined constants in the "Defines" column.

> Example: The rental_methods field contains values CREDITCARD, PAYPASS, etc...

Reviewer: ```suggestion
_Example: The rental_methods field contains values_ CREDITCARD, PAYPASS, etc...
```

* Timestamp - Timestamp fields must be represented as integers in POSIX time. (e.g., the number of seconds since January 1st 1970 00:00:00 UTC)
* ID - Should be represented as a string that identifies that particular entity. An ID:
	* must be unique within like fields (e.g. bike_id must be unique among bikes)
	* does not have to be globally unique, unless otherwise specified
	* must not contain spaces
	* should be persistent for a given entity (station, plan, etc)
* String - Can only contain text. Strings must not contain any formatting codes (including HTML) other than newlines.
* Language - An IETF BCP 47 language code. For an introduction to IETF BCP 47, refer to http://www.rfc-editor.org/rfc/bcp/bcp47.txt and http://www.w3.org/International/articles/language-tags/.

> Examples: en for English, en-US for American English, or de for German.

Reviewer: ```suggestion
Examples: _en_ for English, _en-US_ for American English, or _de_ for German.
```

* Latitude - WGS84 latitude in decimal degrees. The value must be greater than or equal to -90.0 and less than or equal to 90.0.

> Example: 41.890169 for the Colosseum in Rome.

Reviewer: ```suggestion
_Example: 41.890169 for the Colosseum in Rome._
```

* Longitude - WGS84 longitude in decimal degrees. The value must be greater than or equal to -180.0 and less than or equal to 180.0.

> Example: 12.492269 for the Colosseum in Rome.

Reviewer: ```suggestion
_Example: 12.492269 for the Colosseum in Rome._
```

* Non-negative Integer - An integer greater than or equal to 0.
* Non-negative Float - A floating point number greater than or equal to 0.
* Timezone - TZ timezone from the https://www.iana.org/time-zones. Timezone names never contain the space character but may contain an underscore. Refer to http://en.wikipedia.org/wiki/List_of_tz_zones for a list of valid values.

> Example: Asia/Tokyo, America/Los_Angeles or Africa/Cairo.

Reviewer: ```suggestion
_Example: Asia/Tokyo, America/Los_Angeles or Africa/Cairo._
```

```

### gbfs.json

Field Name              | Required    |  Type         | Defines
------------------------| ------------| ------------  | -------------------

> _language_              | Yes         | Language code | The language that will be used throughout the rest of the files. It must match the value in the system_information file.

Reviewer: ```suggestion
_language_              | Yes         | Language | The language that will be used throughout the rest of the files. It must match the value in the system_information file.
```

since we've defined Language as a type, we don't need to add "code"

### system_information.json
The following fields are all attributes within the main "data" object for this feed.

Field Name                      | Required               | Type          | Defines
-----------------------------   | ---------------------  | ----------    | -----------------------
system_id                       | Yes                    | ID            | Identifier for this bike share system. This should be globally unique (even between different systems) - for example,  bcycle_austin or biketown_pdx. It is up to the publisher of the feed to guarantee uniqueness. This value is intended to remain the same over the life of the system.

> language                        | Yes                    | Language code | The language that will be used throughout the rest of the files. It must match the value in the gbfs.json file.

Reviewer: ```suggestion
language                        | Yes                    | Language | The language that will be used throughout the rest of the files. It must match the value in the gbfs.json file.
```

_rental_hours_      | Yes         | Array       | Array of objects as defined below. The array must contain a minimum of one object identifying hours for every day of the week or a maximum of two for each day of the week  objects ( one for each user type).
\-&nbsp;user_types  | Yes         | Array       | An array of "member" and/or "nonmember" value(s). This indicates that this set of rental hours applies to either members or non-members only.
\-&nbsp;days        | Yes         | Array       | An array of abbreviations (first 3 letters) of English names of the days of the week for which this object applies (e.g. ["mon", "tue", “wed”, “thu”, “fri”, “sat”, “sun”]). Rental hours must not be defined more than once for each day and user type.
\-&nbsp;start_time  | Yes         | Time        | Start time for the hours of operation of the system in the time zone indicated in system_information.json.
\-&nbsp;end_time    | Yes         | Time        | End time for the hours of operation of the system in the time zone indicated in system_information.json.

Example:
```json
```
{
  "last_updated": 1434054678,
  "ttl": 0,

>   "version": "1.1",

Reviewer: Undo.

Field Name                      | Required               | Type          | Defines
-----------------------------   | ---------------------  | ----------    | -----------------------
system_id                       | Yes                    | ID            | Identifier for this bike share system. This should be globally unique (even between different systems) - for example,  bcycle_austin or biketown_pdx. It is up to the publisher of the feed to guarantee uniqueness. This value is intended to remain the same over the life of the system.

> language                        | Yes                    | Language      | The language that will be used throughout the rest of the files. It must match the value in the gbfs.json file.

Reviewer: ```suggestion
language                        | Yes                    | Language      | The language that will be used throughout the rest of the files. It must match the value in the [gbfs.json](#gbfsjson) file.
```
